src/program/services/filesystem/vfs/rivenvfs.py

```
@@ -1600,69 +1635,133 @@ def _http_range_request(self, target_url: str, start: int, end: int) -> tuple[in
+                if error_code in (6, 7, 28) and preflight_attempt == 0:  # Host resolution, connection, timeout
+                    fresh_url = self._refresh_download_url(path, target_url)
+
+                    if fresh_url is not None:
+                        target_url = fresh_url
+                        log.info(f"Retrying with fresh URL after connection error for {path}")
+
+                if not is_max_attempt:
+                    await trio.sleep(backoffs[min(preflight_attempt, len(backoffs) - 1)])
+                    continue
+
+                raise pyfuse3.FUSEError(errno.EIO)
```

Ruff (0.13.3)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)